WordPress/src/main/java/org/wordpress/android/ui/pages/PageListFragment.kt

```diff
@@ -54,61 +60,78 @@ class PageListFragment : Fragment() {
+        } else {
+            savedInstanceState.getSerializable(WordPress.SITE) as SiteModel
+        }
+
+        val key = checkNotNull(arguments?.getString(fragmentKey))
 
-        initRecyclerView(savedInstanceState)
+        ifNotNull(activity, siteArgument) { activity, site ->
+            (activity.application as? WordPress)?.component()?.inject(this)
+
+            initializeViews(savedInstanceState)
+            initializeViewModels(activity, site, key)
+        }
     }
 
-    private fun initRecyclerView(savedInstanceState: Bundle?) {
+    override fun onSaveInstanceState(outState: Bundle) {
+        outState.putParcelable(listStateKey, linearLayoutManager.onSaveInstanceState())
+        super.onSaveInstanceState(outState)
+    }
+
+    private fun initializeViewModels(activity: FragmentActivity, site: SiteModel, key: String) {
+        viewModel = ViewModelProviders.of(this, viewModelFactory)
+                .get<PageListViewModel>(key, PageListViewModel::class.java)
+
+        setupObservers()
+
+        val pagesViewModel = ViewModelProviders.of(activity, viewModelFactory).get(PagesViewModel::class.java)
+        viewModel.start(site, getPageType(key), pagesViewModel)
+    }
+
+    private fun initializeViews(savedInstanceState: Bundle?) {
         val layoutManager = LinearLayoutManager(activity, LinearLayoutManager.VERTICAL, false)
         savedInstanceState?.getParcelable<Parcelable>(listStateKey)?.let {
             layoutManager.onRestoreInstanceState(it)
         }
-        recyclerView.layoutManager = linearLayoutManager
-        linearLayoutManager = layoutManager
 
-        val adapter = PagesAdapter { action, pageItem -> viewModel.onAction(action, pageItem) }
-        recyclerView.adapter = adapter
+        linearLayoutManager = layoutManager
+        recyclerView.layoutManager = linearLayoutManager
+        recyclerView.addItemDecoration(RecyclerItemDecoration(0, DisplayUtils.dpToPx(activity, 1)))
+    }
 
-        viewModel.data.observe(this, Observer { data ->
-            if (data != null) {
-                adapter.onNext(data)
-            }
+    private fun setupObservers() {
+        viewModel.pages.observe(this, Observer { data ->
+            data?.let { setPages(data) }
         })
-
-        (activity!!.application as WordPress).component()?.inject(this)
     }
 
-    private fun initViewModel(savedInstanceState: Bundle?) {
-        val key = arguments!!.getString(fragmentKey)
-        viewModel = ViewModelProviders.of(activity!!, viewModelFactory)
-                .get<PageListViewModel>(checkNotNull(key), PageListViewModel::class.java)
-
-        val site = (savedInstanceState?.getSerializable(WordPress.SITE)
-                ?: activity!!.intent!!.getSerializableExtra(WordPress.SITE)) as SiteModel
-
-        viewModel.start(site, key)
+    private fun getPageType(key: String): PageStatus {
+        return when (key) {
+            "key0" -> PageStatus.PUBLISHED
```

Comment:
These strings could be less random (I know I added them like this in the first place 😄 )

Reply:
Good call. I was going to do this but I forgot :).